ir/inference/mod.rs

```diff
@@ -58,5 +37,225 @@ On error, show the inferred types for variables (in minimal original query forma
 In the order that are connected to the variable.
  */
 
-mod type_inference;
-mod value_type_inference;
+use concept::error::ConceptReadError;
+
+pub mod pattern_type_inference;
+mod seed_types;
+pub mod type_inference;
+pub mod value_type_inference;
+
+#[derive(Debug)]
+pub enum TypeInferenceError {
+    ConceptRead { source: ConceptReadError },
+    LabelNotResolved(String),
+}
+
+#[cfg(test)]
```

Comment:
Common functions for tests in `seed_types.rs` and `pattern_type_inference.rs`. It doesn't have to be here.